de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java

```diff
@@ -87,30 +94,46 @@ public void createAndSendDeliveryOrdersForPackages(@NonNull final DeliveryOrderC
 
 	private static DeliveryOrderKey createDeliveryOrderKey(
 			@NonNull final I_M_Package mpackage,
-			final int shipperTransportationId,
-			@NonNull final LocalDate pickupDate)
+			final ShipperTransportationId shipperTransportationId,
+			@NonNull final LocalDate pickupDate,
+			@NonNull final LocalTime timeFrom,
+			@NonNull final LocalTime timeTo)
 	{
 		return DeliveryOrderKey.builder()
-				.shipperId(mpackage.getM_Shipper_ID())
+				.shipperId(ShipperId.ofRepoId(mpackage.getM_Shipper_ID()))
 				.shipperTransportationId(shipperTransportationId)
 				.fromOrgId(mpackage.getAD_Org_ID())
 				.deliverToBPartnerId(mpackage.getC_BPartner_ID())
 				.deliverToBPartnerLocationId(mpackage.getC_BPartner_Location_ID())
 				.pickupDate(pickupDate)
+				.timeFrom(timeFrom)
+				.timeTo(timeTo)
 				.build();
 	}
 
-	private static int computeGrossWeightInKg(final Collection<I_M_Package> mpackages)
+	/**
+	 * In case the weight is <= 0, return the default value.
+	 */
+	private static int computeGrossWeightInKg(@NonNull final Collection<I_M_Package> mpackages, int defaultValue)
 	{
-		return mpackages.stream()
+		final int weightInKg = mpackages.stream()
 				.map(I_M_Package::getPackageWeight) // TODO: we assume it's in Kg
 				.filter(weight -> weight != null && weight.signum() > 0)
 				.reduce(BigDecimal.ZERO, BigDecimal::add)
 				.setScale(0, RoundingMode.UP)
 				.intValueExact();
+
+		if (weightInKg <= 0)
```

Comment:
instead you can use `de.metas.util.lang.CoalesceUtil.firstGreaterThanZero(int...)`

Reply:
Awesome!